File: frontend/src/types/episode.ts
export interface EpisodeProviderPayloadDefaults {
  video: Record<string, unknown>
  tts: Record<string, unknown>
  lipsync: Record<string, unknown>
}

export interface EpisodeWorkflowChecks {
  script_ready: boolean
  providers_ready: boolean
  asset_binding_ready: boolean
  panels_ready: boolean
  voice_ready: boolean
  video_ready: boolean
  compose_ready: boolean
  composed: boolean
}

export interface EpisodeWorkflowCounts {
  required_characters: number
  bound_characters: number
  required_locations: number
  bound_locations: number
  panel_total: number
  panel_video_done: number
  panel_tts_done: number
  panel_lipsync_done: number
}

export interface EpisodeWorkflowSummary {
  current_step: string
  completion_percent: number
  checks: EpisodeWorkflowChecks
  counts: EpisodeWorkflowCounts
  blockers: string[]
  skipped_checks: string[]
}

export interface Episode {
  id: string
  project_id: string
  episode_order: number
  title: string
  summary: string | null
  script_text: string | null
  video_provider_key: string | null
  tts_provider_key: string | null
  lipsync_provider_key: string | null
  provider_payload_defaults: EpisodeProviderPayloadDefaults
  skipped_checks: string[]
  status: string
  panel_count: number
  workflow_summary: EpisodeWorkflowSummary
  created_at: string
  updated_at: string
}

File: backend/app/schemas/episode.py
from __future__ import annotations

from typing import Any

from pydantic import BaseModel


class EpisodeResponse(BaseModel):
    id: str
    project_id: str
    episode_order: int
    title: str
    summary: str | None
    script_text: str | None
    video_provider_key: str | None
    tts_provider_key: str | None
    lipsync_provider_key: str | None
    provider_payload_defaults: dict[str, dict[str, Any]]
    skipped_checks: list[str]
    status: str
    panel_count: int = 0
    workflow_summary: dict[str, Any]
    created_at: str
    updated_at: str
